packages/keyring/src/keyring.ts

```diff
@@ -0,0 +1,333 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressHash } from '@alephium/shared'
+import { addressToGroup, bs58, ExplorerProvider, sign, TOTAL_NUMBER_OF_GROUPS, transactionSign } from '@alephium/web3'
+import { generateMnemonic, mnemonicToSeedSync, validateMnemonic } from '@metamask/scure-bip39'
+import { wordlist } from '@metamask/scure-bip39/dist/wordlists/english'
+import blake from 'blakejs'
+import { HDKey } from 'ethereum-cryptography/hdkey'
+import { bytesToHex } from 'ethereum-cryptography/utils'
+
+import { decryptMnemonic, DecryptMnemonicResult, MnemonicLength, mnemonicStringToUint8Array } from '@/mnemonic'
+
+export type NonSensitiveAddressData = {
+  hash: AddressHash
+  index: number
+  publicKey: string
+}
+
+type SensitiveAddressData = NonSensitiveAddressData & {
+  privateKey: Uint8Array
+}
+
+type GenerateAddressProps = {
+  group?: number
+  addressIndex?: number
+  skipAddressIndexes?: number[]
+}
+
+class Keyring {
+  private hdPath = "m/44'/1234'/0'/0"
+  private hdWallet: HDKey | null
+  private root: HDKey | null
+  private addresses: SensitiveAddressData[]
+
+  constructor() {
+    this.addresses = []
+    this.hdWallet = null
+    this.root = null
+  }
+
+  // PUBLIC METHODS
+
+  public clearCachedSecrets = () => {
+    this.addresses = []
+    this.hdWallet = null
+    this.root = null
+  }
+
+  public generateRandomMnemonic = (mnemonicLength: MnemonicLength = 24): Uint8Array => {
+    // const strength = mnemonicLength === 24 ? 256 : 128
+    const mnemonic = generateMnemonic(wordlist)
```

Comment:
The `generateMnemonic` function of `@scure/bip39` does not accept mnemonic length as the one from `bip39` does.